indico/modules/events/registration/client/js/form/fields/DateInput.jsx

```diff
@@ -42,18 +44,16 @@ export function DateInputComponent({
   return (
     <Form.Group styleName="date-field">
       <Form.Field>
-        <SingleDatePicker
+        <Datepicker
           id={id}
           name="date"
           disabled={disabled}
           required={required}
-          date={toMoment(dateValue, 'YYYY-MM-DD', true)}
-          onDateChange={handleDateChange}
-          placeholder={dateFormat}
-          displayFormat={dateFormat}
-          isOutsideRange={() => false}
-          verticalSpacing={10}
-          enableOutsideDays
+          format={dateFormat}
+          value={dateValue}
```

Comment:
FWIW the way I would expect the Datepicker component to work is this:

- The `value` it receives is always in ISO format
- The `value` it sends (via `onChange`) is always in ISO format
- The displayed value, placeholder and input format (when manually entering a date) uses the specified `format` (and I think ideally that `format` would be optional and default to the locale's date format since the regform is the only place where we want to override the format instead of using the locale's default)

Reply:
Gotcha. What's the best place to obtain the locale-format mapping?

Reply:
I just discussed w/ @tomasr8 and maybe using momentjs here would be the best option (`moment(...).format('L')`) - we still use it in many places so it's not likely to go away any time soon (even though Intl nowadays supports pretty much everything).

The moment docs also indicate that this is always a number-based format, so it'll probably be fine for all locales:

> Month numeral, day of month, year | L | 09/04/1986

Reply:
`moment.localeData().longDateFormat('L')`

Reply:
Looks like we can get that information from `moment.localeData().longDateFormat('L')`

Reply:
Not sure why this was marked as resolved, I still get this behavior...

Reply:
I think the problem is that currently the value is going directly into the `defaultValue` of the text input field. Maybe it should be passed e.g. as `<ind-date-picker defaultValue={value}>` instead, so it can be converted to the proper format before it's put in the input field?

Reply:
I resolved this by mistake. Anyway, fixed the issue by reformatting the date prior to using it as the default value.